### PR TITLE
rtaudio: add recipe

### PR DESCRIPTION
**Pull request by hfsfox**
No mediakit backend at the moment, dummy output only

### CODE REVIEW

**Comment by korli** [comment]
Should be disabled if it doesn't work.

**Comment by hfsfox** [comment]
@korli it is required to build other applications such as bambootracker

**Comment by korli** [comment]
@hfsfox will bambootracker be usable without sound output?

**Comment by hfsfox** [comment]
@korli yes, because midi via rtmidi works fine, and rtaudio is the only thing required for build now.

**Comment by diversys** [comment]
> @korli yes, because midi via rtmidi works fine, and rtaudio is the only thing required for build now.

Makes sense. @korli seems like it's good to go?

File: media-libs/rtaudio/patches/rtaudio-5.2.0.patchset [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   list(APPEND LINKLIBS Threads::Threads)
+ endif()
+ 
+@@ -315,7 +319,11 @@ install(FILES RtAudio.h rtaudio_c.h
+ export(PACKAGE RtAudio)
+ 
+ # Set installation path for CMake files.
+-set(RTAUDIO_CMAKE_DESTINATION share/rtaudio)
++if (HAIKU)
++    set(RTAUDIO_CMAKE_DESTINATION lib/cmake/rtaudio)
```
Comment (Begasus) [inline media-libs/rtaudio/patches/rtaudio-5.2.0.patchset:70]: Maybe (haven't check, but cmake should end up in lib/x86 on 32bit)?
```diff
-+    set(RTAUDIO_CMAKE_DESTINATION lib/cmake/rtaudio)
++    set(RTAUDIO_CMAKE_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/rtaudio)
```

**Comment by korli** [comment]
> > @korli yes, because midi via rtmidi works fine, and rtaudio is the only thing required for build now.
> 
> Makes sense. @korli seems like it's good to go?

IMO rtaudio shouldn't be required, if the software is usable without it.

**Comment by Begasus** [comment]
https://github.com/BambooTracker/BambooTracker/releases/download/v0.6.2/BambooTracker-src-v0.6.2.tar.gz includes the sources for rtaudio and rtmidi, system rtmidi can be used as a cmake variable.
From upstream:
```
RtAudio / RtMidi: Use a precompiled RtAudio / RtMidi installation instead of our vendored one.
(Obsoletes the optional dependencies above)
Requires RtAudio (5.1.0 or higher) and/or RtMidi (4.0.0 or higher) + pkg-config.
```
So I'm guessing it's not a hard requirement (can build it without rtaudio).
Could you check this forr BoomTracker @hfsfox ?

**Comment by Begasus** [comment]
BambooTracker builds fine with: 

`cmake -Bbuild -S. -DCMAKE_INSTALL_PREFIX=/boot/home/destdir/ -DRTMIDI_API_ALSA=OFF`

Can't seem to do anything else with it though. Doesn't even open the demo files.

**Comment by Begasus** [comment]
@hfsfox could you mark the arch's as "?", when enabled it would lead to false ideas thinking this is fully functional and will flood the forum probably with question as why there is no output. :)
Once done this could be merged and one that needs it can build it themselves to go from here.

**Comment by Begasus** [comment]
Merging disabled for now.

**Pull request merged**